article/app/services/dotcomponents/ArticlePicker.scala

```diff
@@ -4,7 +4,8 @@ import controllers.ArticlePage
 import experiments.{ActiveExperiments, DotcomRenderingAdvertisements}
 import model.PageWithStoryPackage
 import implicits.Requests._
-import model.liveblog.{BlockElement, ImageBlockElement, PullquoteBlockElement, TextBlockElement, TweetBlockElement, RichLinkBlockElement}
+import model.dotcomrendering.pageElements.{BlockquoteBlockElement, SubheadingBlockElement, YoutubeBlockElement}
```

Comment:
@tjmw does this look right to you?

Reply:
Hey, I'm not sure. I'm not super familiar with this code unfortunately. One thing I'm slightly confused about is that the `unsupportedElement` function below says that it takes a `BlockElement` and then we match on the specific type. But those new entries aren't `BlockElement`s, they're `PageElements`s and are defined separately. I'm probably getting the wrong end of the stick somewhere here (especially if you've seen this work 😛) but I'd be interested to learn more about what's happening here.

Reply:
Yeah, I think you're right and it's not quite right!

Reply:
Huh yeah definitely weird! Conceptually it makes total sense (i.e. that we would base the check on which elements are supported on the models available under `model.dotcomrendering.pageElements`). Let me know if I can help with figuring this out!

Reply:
I'd definitely double check this code. It looks like importing pageElements stuff is something introduced on this PR, so I'd be surprised if that's what it's meant to do.

--

Some historical notes about page vs block elements, *this is from memory so don't 100% assume this is correct*

The first pieces of code in frontend written to use the block/element model was the liveblog page (by Simon!), which is why they all live underneath model.liveblog.

pageElements was a concept introduced for DCR. The idea was that we didn't want to touch the original set of data models so we created a whole new set of them. So we could be certain we wouldn't screw up the original site accidentally.

I think that the picker runs *before we process the capi model into the DCR model* so they won't be pageElement types yet, so it should be working off of the model.liveblog stuff. That being said, intellij has a habit of trying to auto import them, not sure why...

Reply:
@MatthewJWalls there's still space on the desk for you......

Reply:
![](https://i.makeagif.com/media/5-21-2017/RXjkJn.gif)

Reply:
Blast from the past! Thanks Matt, you must be really bored. ❤️